orchagent/vnetorch.cpp

```diff
@@ -3039,3 +3139,179 @@ bool MonitorOrch::delOperation(const Request& request)
+
+    vector<FieldValueTuple> fvs = {
+        {ACL_TABLE_TYPE_MATCHES, matches},
+        {ACL_TABLE_TYPE_ACTIONS, ACTION_REDIRECT_ACTION},
+        {ACL_TABLE_TYPE_BPOINT_TYPES, bpoints}
+    };
+
+    acl_table_type_->set(VNET_TUNNEL_TERM_ACL_TABLE_TYPE, fvs);
+    
+    std::string ports_str = "";
+    auto ports = getBindPoints();
+    if (!ports.empty())
+    {
+        ports_str = std::accumulate(std::next(ports.begin()), ports.end(), ports[0], concat);
+    }
+
+    vector<FieldValueTuple> fvs2 = {
+        {ACL_TABLE_DESCRIPTION, "Vnet Tunnel Termination ACL"},
+        {ACL_TABLE_TYPE, VNET_TUNNEL_TERM_ACL_TABLE_TYPE},
+        {ACL_TABLE_STAGE, STAGE_INGRESS},
+        {ACL_TABLE_PORTS, ports_str}
+    };
+
+    acl_table_->set(VNET_TUNNEL_TERM_ACL_TABLE, fvs2);
+
+    acl_table_initialized_ = true;
+}
+
+std::vector<std::string> VNetTunnelTermAcl::getBindPoints()
+{
+    std::vector<std::string> bind_points;
+    std::set<std::string> internal_ports = findInternalPorts();
+    auto all_ports = gPortsOrch->getAllPorts();
+
+    std::set<std::string> legitSet;
+
+    /* Add physical and LAG prots */
+    for (auto& it : all_ports)
+    {
+        Port& port = it.second;
+        if (port.m_type == Port::PHY || port.m_type == Port::LAG)
+        {
+            legitSet.insert(it.first);
+        }
+    }
+
+    /* Remove LAG members */
+    for (auto& it : all_ports)
+    {
+        Port& port = it.second;
+        if (port.m_type == Port::LAG)
+        {
+            for (auto member : port.m_members)
+            {
+                legitSet.erase(member);
+            }
+        }
+    }
+
+    for (auto& port : legitSet)
+    {
+        if (internal_ports.find(port) == internal_ports.end())
+        {
+            bind_points.push_back(port);
+        }
+    }
+
+    return bind_points;
+}
+
+std::set<std::string> VNetTunnelTermAcl::findInternalPorts()
+{
+    std::set<std::string> internal_ports;
+    std::vector<std::string> all_ports;
+    port_table_->getKeys(all_ports);
+
+    for (auto& port : all_ports)
+    {
+        std::string role;
+        if (port_table_->hget(port, PORT_ROLE, role))
+        {
+            if (role == PORT_ROLE_DPC)
+            {
+                internal_ports.insert(port);
+            }
+        }
+    }
+
+    return internal_ports;
+}
+
+bool VNetTunnelTermAcl::createAclRule(const string vnet_name, swss::IpPrefix& vip, swss::IpAddress nh_ip)
+{
+    SWSS_LOG_ENTER();
+
+    lazyInit();
+
+    std::string rule_name = getRuleName(vnet_name, vip);
+    std::string alias = getNbrAlias(nh_ip);
+
+    if (alias.empty())
+    {
+        SWSS_LOG_ERROR("Failed to get interface alias for IP %s", nh_ip.to_string().c_str());
+        return false;
+    }
+
+    vector<FieldValueTuple> fvs = {
+        {RULE_PRIORITY, to_string(VNET_TUNNEL_TERM_ACL_BASE_PRIORITY)},
+        {MATCH_DST_IP, vip.to_string()},
+        {ACTION_REDIRECT_ACTION, alias}
```

Comment:
Hi, Why is the redirect set to an alias and not IP. For ENI based forwarding, we set it to prefix.

Reply:
So, both the DPU's has same VIP?

Reply:
Correct.